### PR TITLE
[Core][State Observability] Refactor ray logs CLI semantics 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This PR changes `ray logs [<multiple options mixed for various entities>]` to `ray logs file ...` and `ray logs actor ...` 

This is a result of preliminary feedback on the [API spec](https://docs.google.com/document/d/1eyvSPnYgXBdEXB2-qm_gKDDfO2eEnHe7R1Mu6y-9FWU/edit#heading=h.136e81d1cyao)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_state_api_log.py
Comment (rickyyx) [inline python/ray/tests/test_state_api_log.py:605]: This is why the accidentally merged PR didn't fail the test. I think this is actually a bug, i.e, how to handle nodes with same node-ip but different node-ids.

Comment (rkooo567) [reply python/ray/tests/test_state_api_log.py:605]: Can you create an issue?

File: doc/source/ray-observability/state/ray-state-api-reference.rst
Comment (rkooo567) [inline doc/source/ray-observability/state/ray-state-api-reference.rst:40]: We also need ray logs worker <Worker ID OR PID>

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:699]: Is it possible to do `ray logs raylet.out` too? If so, can you mention `ray logs` is a alias of `ray logs file`?

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:716]:
```suggestion
    """Parse the query API server and the target nodes from which logs should be retrieved.
```

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:715]:
```suggestion
) -> Tuple[str, str, str]:
```

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:722]: Returns:
    Tuple of API server URL, node id, and node ip. If no node id and node ip is provided, it will return the head node ip

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:974]: Isn't it 
`ray logs actor --pid=123 --ip=ABC`?

File: python/ray/experimental/state/state_cli.py
Comment (rkooo567) [inline python/ray/experimental/state/state_cli.py:800]: We are missing

```
        if not tail:
            tail = 0 if follow else DEFAULT_LOG_LIMIT
```
?

File: doc/source/ray-observability/state/ray-state-api-reference.rst
Comment (rkooo567) [reply doc/source/ray-observability/state/ray-state-api-reference.rst:40]: Maybe we can do it in a follow up

**Review by rkooo567 — APPROVED** [review]
Mostly nit comments

File: doc/source/ray-observability/state/state-api.rst
Comment (rkooo567) [inline doc/source/ray-observability/state/state-api.rst:266]: Can you change the order of examples actually? We should bring up high level API before the low level API (so ray logs actor & ray logs worker is more important than ray logs file)

**Comment by rickyyx** [comment]
I think I will wait until #26997 merged first, there will be quite some conflicts.

**Comment by rkooo567** [comment]
We need to discuss if we can pick this to the 2.0 branch. Let's talk about it later

**Comment by rkooo567** [comment]
We decided to pick to avoid sudden API changes from alpha -> beta. I think the risk is low. But before we merge this @rickyyx can you play with the wheel to make sure it doesn't introduce unexpected regression ?

**Comment by rickyyx** [comment]
> We decided to pick to avoid sudden API changes from alpha -> beta. I think the risk is low. But before we merge this @rickyyx can you play with the wheel to make sure it doesn't introduce unexpected regression ?

Will do today.

**Comment by rickyyx** [comment]
I will pull #26997 into this PR, and build on top of that.

File: doc/source/ray-observability/state/ray-state-api-reference.rst
Comment (rickyyx) [reply doc/source/ray-observability/state/ray-state-api-reference.rst:40]: But we don't have `--worker-id` yet? Is this working already?

File: python/ray/experimental/state/state_cli.py
Comment (rickyyx) [reply python/ray/experimental/state/state_cli.py:800]: Why would we want to set `tail=0` when `follow=True`?

`tail` will never be None with default values now.

**Comment by rickyyx** [comment]
> I will pull #26997 into this PR, and build on top of that.

Hmm, seems this workflow doesn't work that well as the changes from #26997 are also here...

**Comment by scv119** [comment]
remove from v2.0.0-pick as no activities for ~10 days

**Comment by rickyyx** [comment]
Deferring this PR due to ongoing discussion about how to integrate with anyscale. Revisit after requirements are clearer.

**Pull request closed without merging**
